src/server/firefox.ts

```diff
@@ -87,21 +87,30 @@ export class Firefox extends AbstractBrowserType<FFBrowser> {
     assert(!port || launchType === 'server', 'Cannot specify a port without launching as a server.');
     const logger = new RootLogger(options.logger);
 
-    const firefoxArguments = [];
-
     let temporaryProfileDir = null;
     if (!userDataDir) {
       userDataDir = await mkdtempAsync(path.join(os.tmpdir(), 'playwright_dev_firefox_profile-'));
       temporaryProfileDir = userDataDir;
     }
 
+    let firefoxArguments = [];
     if (!ignoreDefaultArgs)
       firefoxArguments.push(...this._defaultArgs(options, launchType, userDataDir, 0));
     else if (Array.isArray(ignoreDefaultArgs))
       firefoxArguments.push(...this._defaultArgs(options, launchType, userDataDir, 0).filter(arg => !ignoreDefaultArgs.includes(arg)));
     else
       firefoxArguments.push(...args);
 
+    const urls = firefoxArguments.filter((arg, index, args) => !arg.startsWith('-') && args[index - 1] !== '-profile' && args[index - 1] !== '-juggler');
```

Comment:
It might be time to depend on minimist in a prod mode. You are throwing on various valid params.